AsyncDisplayKit/ASButtonNode.mm

```diff
@@ -480,7 +481,11 @@ - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
   }
   
   if (_titleNode.attributedText.length > 0) {
-    [children addObject:_titleNode];
+    if (_imageAlignment == ASButtonNodeImageAlignmentBeginning) {
+      [children addObject:_titleNode];
+    } else {
+      [children insertObject:_titleNode atIndex:0];
```

Comment:
This could certainly be changed later, but it's a bit unintuitive that the image alignment option would actually affect the location of the _other_ nodes in the stack.